### PR TITLE
Configurable garbage collection of storage manager metadata

### PR DESCRIPTION
**Pull request by sreeja**

### CODE REVIEW

**Comment by Mallets** [comment]
Please add config example in `https://github.com/eclipse-zenoh/zenoh/blob/master/DEFAULT_CONFIG.json5`

File: DEFAULT_CONFIG.json5 [hunk 1]
```diff
@@ -297,6 +297,17 @@
 //        demo2: {
 //          key_expr: "demo/memory2/**",
 //          volume: "memory",
+//          /// Storage manager plugin handles metadata in order to ensure convergence of distributed storages configured in Zenoh.
+//          /// Metadata includes the set of wild card updates and deletions (tombstones).
+//          /// Once the samples are guaranteed to be delivered, the metadata can be garbage collected.
+//          garbage_collection_config: {
```
Comment (Mallets) [inline DEFAULT_CONFIG.json5:303]: I would call it simply *garbace_collection*.

File: DEFAULT_CONFIG.json5 [hunk 2]
```diff
@@ -297,6 +297,17 @@
...
+//            /// The garbage collection event will be periodic with this duration.
+//            /// The duration should be specified in seconds.
+//            garbage_collection_period: 30,
```
Comment (Mallets) [inline DEFAULT_CONFIG.json5:306]: We are already in the *garbage_collection* section, let's simply call it *period*.

File: DEFAULT_CONFIG.json5 [hunk 3]
```diff
@@ -297,6 +297,17 @@
...
+//            /// Metadata older than this parameter will be garbage collected.
+//            /// The duration should be specified in seconds.
```
Comment (Mallets) [inline DEFAULT_CONFIG.json5:308]: The duration **IS** specified in seconds.

File: DEFAULT_CONFIG.json5 [hunk 4]
```diff
@@ -297,6 +297,17 @@
 //        demo2: {
 //          key_expr: "demo/memory2/**",
 //          volume: "memory",
+//          /// Storage manager plugin handles metadata in order to ensure convergence of distributed storages configured in Zenoh.
+//          /// Metadata includes the set of wild card updates and deletions (tombstones).
+//          /// Once the samples are guaranteed to be delivered, the metadata can be garbage collected.
+//          garbage_collection_config: {
+//            /// The garbage collection event will be periodic with this duration.
+//            /// The duration should be specified in seconds.
```
Comment (Mallets) [inline DEFAULT_CONFIG.json5:305]: The duration **IS** specified in seconds.

File: plugins/zenoh-backend-traits/src/config.rs [hunk 5]
```diff
@@ -76,6 +77,25 @@ impl Default for ReplicaConfig {
     }
 }
 
+// The configuration for periodic garbage collection of metadata in storage manager
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct GarbageCollectionConfig {
+    // The duration between two garbage collection events
+    // The garbage collection will be scheduled as a periodic event with this period
+    pub garbage_collection_period: Duration,
```
Comment (Mallets) [inline plugins/zenoh-backend-traits/src/config.rs:85]: Just call the field *period*.

File: plugins/zenoh-backend-traits/src/config.rs [hunk 6]
```diff
@@ -76,6 +77,25 @@ impl Default for ReplicaConfig {
...
+    // The metadata older than this parameter will be garbage collected
+    pub garbage_collection_delay: Duration,
```
Comment (Mallets) [inline plugins/zenoh-backend-traits/src/config.rs:87]: Just call the field ?*age*?

Comment (sreeja) [reply plugins/zenoh-backend-traits/src/config.rs]: I renamed it as _lifespan_ which I believe is slightly better than _age_. Do you agree?

File: DEFAULT_CONFIG.json5 [hunk 7]
```diff
@@ -297,6 +297,17 @@
...
+//            garbage_collection_period: 30,
+//            /// Metadata older than this parameter will be garbage collected.
+//            /// The duration should be specified in seconds.
+//            garbage_collection_delay: 86400,
```
[the remark this answers is not in the document]
Comment (sreeja) [reply DEFAULT_CONFIG.json5]: I renamed it as _lifespan_ which I believe is slightly better than _age_. Do you agree?

[the remark this answers is not in the document]
Comment (Mallets) [reply DEFAULT_CONFIG.json5]: I agree

File: plugins/zenoh-backend-traits/src/config.rs (hunk 6 above)
Comment (Mallets) [reply plugins/zenoh-backend-traits/src/config.rs]: I agree

**Review by Mallets — APPROVED** [review]

**Pull request merged**